### PR TITLE
Switch to using an object for the arguments and accept a new `arpPath` argument

### PR DESCRIPTION
**Pull request by timrogers**
This updates the API of the `findLocalDevices` function to allow you to optionally specify the path of the `arp` executable using the `arpPath` option, rather than assuming that it is available in the `PATH`.

As part of this change, I make a **breaking change** to the API of the function, switching from positional arguments to a keyed object. This makes the API more friendly and discoverable as the number of options grows, and means you don't have to use `null`s to say "I don't want to set this option".

It also corrects the `findLocalDevices` function type to reflect the `skipNameResolution` argument which is already accepted.

### CODE REVIEW

**Comment by DylanPiercey** [comment]
@timrogers i think this looks good. Could you update the docs?

File: src/index.d.ts
Comment (natterstefan) [inline src/index.d.ts:3]:
```suggestion
    function findLocalDevices(address?: any, { skipNameResolution?: boolean, arpPath?: string }): Promise<findLocalDevices.IDevice[]>;
```

I vote for adding the additional parameters in an `options` object. This allows us to add more flags, options, etc. in the future without breaking the API or extending the list of parameters much longer.

@DylanPiercey This will be breaking, _but_ will allow us to be more flexible in the future!

### Example

```js
const foo = findLocalDevices('192.168.0.10', {skipNameResolution: false, arpPath: 'arp'})
```

Two more things

- take care in the test that this is tested
- update the README accordingly.

Thanks.

PS: please change the type of `address` to `address?: string | null`.

**Review by natterstefan — CHANGES_REQUESTED** [review]

File: src/index.d.ts
Comment (DylanPiercey) [reply src/index.d.ts:3]: Yeah I agree.

Comment (timrogers) [reply src/index.d.ts]: Definitely agree on this - it's a breaking change, but it makes a lot of sense once you go beyond a couple of arguments.

I've implemented the proposed change, and actually gone a bit further - I've moved the `address` argument into the object and renamed it to `ipAddressOrRange`. This feels like the right move to me - it's weird to have an optional (nullable?) positional argument followed by an object.

How do these updates look to you? 😊

**Comment by timrogers** [comment]
@DylanPiercey 👋🏻 Hope you're having a good weekend! Any chance we can get this merged?

**Comment by timrogers** [comment]
@DylanPiercey 👋🏻 Just a reminder about this PR when you have a second!

**Comment by DylanPiercey** [comment]
Hey @timrogers @natterstefan! I'm having a hard time maintaining this project since I haven't actively used it in years and I don't have a lot of confidence in its test suite (eg we're not testing against windows).

Wondering if either of you would like to step in as a maintainer on GitHub (I know @natterstefan already is) with npm publish access also?

**Comment by timrogers** [comment]
I would be happy to take it on. I don't realistically have huge amounts of time to drive the project forward, but I can at least steward it and review things.

File: src/index.d.ts
Comment (DylanPiercey) [reply src/index.d.ts]: How about just `address` for the name? `ipAddressOrRange` seems a bit verbose.

Comment (timrogers) [reply src/index.d.ts]: I'm cool with that. We can use the description to make it clear that it can be a single address or a range.

Comment (DylanPiercey) [reply src/index.d.ts]: @timrogers if you want to go ahead with that then that'd be great. I think otherwise I'm happy with the changes!

File: src/index.d.ts
Comment (DylanPiercey) [inline src/index.d.ts:3]: I think this type needs to be updated to show that `opts` is optional (`opts?`).

Comment (timrogers) [reply src/index.d.ts:3]: Very good spot!

**Comment by natterstefan** [comment]
Hey @DylanPiercey,
Hey @timrogers,

> Wondering if either of you would like to step in as a maintainer on GitHub (I know @natterstefan already is) with npm publish access also?

I'm sorry I missed lots of PRs too. Additionally, I was not exactly sure how we proceed with PRs (do we both need to accept it or just one of us) and how we proceed with releases after merges. I am happy to step in as a more active maintainer (together with @timrogers if he wants) proceeding more actively PRs etc. up to a release on npm.

> I don't have a lot of confidence in its test suite (eg we're not testing against windows).

Me too with the windows part, but I could set up at least GitHub actions for Linux, macOS, and Windows and execute the command in a native environment.

**Comment by timrogers** [comment]
@DylanPiercey Ready for re-review!

**Comment by DylanPiercey** [comment]
Looks good, thanks!

**Comment by DylanPiercey** [comment]
Published as `4.0.0`

**Comment by natterstefan** [comment]
Thank you, @DylanPiercey.

**Pull request merged**
